eng/common/scripts/FilterPoliCheckResults.ps1

```diff
@@ -6,35 +6,42 @@ This script will read data speciefied in one or more PoliCheckAllowList.yml file
 It then reamoves all allwed entries from the PoliCheckResult 
 .PARAMETER PoliCheckResultFilePath
 The Path to the PoliCheck Result. Usually named PoliCheck.sarif
-.PARAMETER ServiceDirtectory
+.PARAMETER ServiceDirectory
 If the PoliCheck scan is scoped to a particular service provide the ServiceDirectory
+.PARAMETER AllowListLocation
+A path to a folder containing yml defined policheck allowlist
 .EXAMPLE
-PS> ./FilterPoliCheckResults.ps1 -PoliCheckResultFilePath .\PoliCheck.sarif
+PS> ./FilterPoliCheckResults.ps1 -PoliCheckResultFilePath .\PoliCheck.sarif -ServiceDirectory <servicedirname> -AllowListLocation <location>
 #>
 [CmdletBinding()]
 param(
   [Parameter(Mandatory=$true)]
   [String] $PoliCheckResultFilePath,
-  [String] $ServiceDirtectory
+  [Parameter(Mandatory=$true)]
+  [String] $ServiceDirectory,
+  [Parameter(Mandatory=$true)]
+  [String] $AllowListLocation
 )
 
 . "${PSScriptRoot}\logging.ps1"
 
-$RepoRoot = Resolve-Path -Path "${PSScriptRoot}\..\..\..\"
-$PathToAllowListFiles = Join-Path $RepoRoot $ServiceDirtectory
-$PolicCheckAllowListFiles = Get-ChildItem -Path $PathToAllowListFiles -Recurse -File -Include "PoliCheckAllowList.yml"
+# Install Powershell Yaml
+$ProgressPreference = "SilentlyContinue"
+$toolsFeed = "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-tools/nuget/v2"
+Register-PSRepository -Name azure-sdk-tools-feed -SourceLocation $toolsFeed -PublishLocation $toolsFeed -InstallationPolicy Trusted -ErrorAction SilentlyContinue
+Install-Module -Repository azure-sdk-tools-feed powershell-yaml
+
+$allowListFilePath = Join-Path $AllowListLocation "${ServiceDirectory}.yml"
 $allowListData = @{}
 
-# Combine all AllowLists Found
-foreach ($file in $PolicCheckAllowListFiles)
+if (Test-Path -Path $allowListFilePath)
 {
-    $allowListDataInFile = ConvertFrom-Yaml (Get-Content $file.FullName -Raw)
-    $allowListData["PC1001"] += $allowListDataInFile["PC1001"]
-    $allowListData["PC1002"] += $allowListDataInFile["PC1002"]
-    $allowListData["PC1003"] += $allowListDataInFile["PC1003"]
-    $allowListData["PC1004"] += $allowListDataInFile["PC1004"]
-    $allowListData["PC1005"] += $allowListDataInFile["PC1005"]
-    $allowListData["PC1006"] += $allowListDataInFile["PC1006"]
+    $allowListData = ConvertFrom-Yaml (Get-Content $allowListFilePath -Raw)
+}
+else
+{
+    LogError "Allow list path ${allowListFilePath} does not exisit."
```

Comment:
```suggestion
    LogError "Allow list path ${allowListFilePath} does not exist."
```

Reply:
Is this gonna work. Won't $allowListFilePath just pring as a literal.

Reply:
```
$ $foo = "foostring"
$ Write-Host "the value is $foo"
the value is foostring
```

Curly braces are only needed in cases where there isn't a clear word boundary between the variable and subsequent characters, or where there are special characters in the variable name.

https://docs.microsoft.com/en-us/powershell/scripting/learn/deep-dives/everything-about-string-substitutions?view=powershell-7.1#delineation-with-braces